cloudinit/sources/helpers/hetzner.py

```diff
@@ -3,22 +3,45 @@
 #
 # This file is part of cloud-init. See LICENSE file for license information.
 
-from cloudinit import url_helper, util
+from typing import Optional, Tuple
 
+from cloudinit import net, url_helper
 
-def read_metadata(url, timeout=2, sec_between=2, retries=30):
-    response = url_helper.readurl(
-        url, timeout=timeout, sec_between=sec_between, retries=retries
-    )
-    if not response.ok():
-        raise RuntimeError("unable to read metadata at %s" % url)
-    return util.load_yaml(response.contents.decode(), allowed=(dict, list))
 
+def skip_retry_on_empty_response(cause: url_helper.UrlError) -> bool:
```

Comment:
Nit: This function is not used anywhere, we could make it private to the module by: `_skip_retry_on_empty_response`.